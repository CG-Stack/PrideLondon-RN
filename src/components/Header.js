// @flow
import React from "react";
import type { Node } from "react";
import { StatusBar } from "react-native";
import SafeAreaView from "react-native-safe-area-view";

type Props = {
  backgroundColor: string,
  children: Node
};

const Header = ({ backgroundColor, children }: Props) => (
  <SafeAreaView accessibilityTraits={["header"]} style={{ backgroundColor }}>
    <StatusBar
      barStyle="light-content"
      animated
      backgroundColor={backgroundColor}
    />
    {children}
  </SafeAreaView>
);

export default Header;
