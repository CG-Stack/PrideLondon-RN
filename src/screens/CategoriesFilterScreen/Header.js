// @flow
import React from "react";
import { View, TouchableOpacity, StyleSheet } from "react-native";
import Text from "../../components/Text";
import text from "../../constants/text";
import { whiteColor } from "../../constants/colors";
import CategoriesPills from "../../components/CategoriesPills";

type Props = {
  onClose?: Function,
  onClearAll?: Function,
  selectedCategories: Set<string>
};

type ActionButtonProps = {
  label: string,
  onPress?: Function
};

const ActionButton = ({ label, onPress }: ActionButtonProps) => (
  <TouchableOpacity style={styles.actionButton} onPress={onPress}>
    <Text type="h4" style={styles.actionButtonText}>
      {label}
    </Text>
  </TouchableOpacity>
);

ActionButton.defaultProps = {
  onPress: () => {}
};

const Header = ({ onClose, onClearAll, selectedCategories }: Props) => (
  <View>
    <View style={styles.actionButtons}>
      <ActionButton label={text.cancel} onPress={onClose} />
      {selectedCategories.size > 0 && (
        <ActionButton label={text.clearAll} onPress={onClearAll} />
      )}
    </View>
    <Text type="h1" style={styles.filterTitle}>
      {text.filterTitle}
    </Text>
    <CategoriesPills
      style={styles.categoriesPills}
      selectedCategories={selectedCategories}
    />
  </View>
);

Header.defaultProps = {
  onClose: () => {},
  onClearAll: () => {}
};

const styles = StyleSheet.create({
  actionButtons: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 10
  },
  actionButton: {
    width: 60
  },
  actionButtonText: {
    color: whiteColor
  },
  categoriesPills: {
    marginBottom: 16
  },
  filterTitle: {
    color: whiteColor,
    marginTop: 14,
    marginBottom: 6
  }
});

export default Header;
