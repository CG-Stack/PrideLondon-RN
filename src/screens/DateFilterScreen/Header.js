// @flow
import React from "react";
import { View } from "react-native";
import CommonHeader from "../../components/Header";
import ActionButton from "../../components/ActionButton";

type Props = {
  onCancel: () => void,
  onReset: () => void,
  dateRange: string,
  title: string,
  titleLabel: string
};

const Header = ({ onCancel, onReset, dateRange, title, titleLabel }: Props) => (
  <View>
    <CommonHeader
      leftElement={<CommonHeader.BackButton onPress={onCancel} />}
      title={title}
      titleLabel={titleLabel}
      titleAccesible
      titleAccessibilityTraits={["header"]}
      titleAccessibilityLiveRegion="polite"
      rightElement={
        dateRange && <ActionButton label="Reset" onPress={onReset} />
      }
    />
  </View>
);

export default Header;
