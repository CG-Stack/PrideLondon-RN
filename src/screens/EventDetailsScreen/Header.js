// @flow
import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import SafeAreaView from "react-native-safe-area-view";

type HeaderProps = {
  onBackButtonPress: () => void
};

const Header = ({ onBackButtonPress }: HeaderProps) => (
  <SafeAreaView>
    <View style={styles.header}>
      <TouchableOpacity onPress={onBackButtonPress}>
        <Text>Back</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  header: {
    height: 180,
    padding: 22
  }
});

export default Header;
