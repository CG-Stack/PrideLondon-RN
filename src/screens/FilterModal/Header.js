// @flow
import React from "react";
import ActionButton from "../../components/ActionButton";
import GlobalHeader from "../../components/Header";
import text from "../../constants/text";

type Props = {
  onClearPress: () => void,
  onCancelPress: () => void,
  showClear: boolean
};

const Header = ({ onClearPress, onCancelPress, showClear }: Props) => {
  const clearElement = showClear ? (
    <ActionButton label={text.clearAll} onPress={onClearPress} />
  ) : null;
  return (
    <GlobalHeader
      leftElement={<ActionButton label={text.cancel} onPress={onCancelPress} />}
      title={text.filterEvents}
      rightElement={clearElement}
    />
  );
};

export default Header;
