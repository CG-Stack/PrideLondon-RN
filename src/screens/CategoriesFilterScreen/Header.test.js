// @flow
import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

const render = props => shallow(<Header {...props} />);

describe("Header Component", () => {
  it("renders correctly", () => {
    const output = render({ selectedCategories: new Set() });
    expect(output).toMatchSnapshot();
  });

  it("does not render clear all button if there are no selected categories", () => {
    const output = render({ selectedCategories: new Set() });

    expect(output.find('ActionButton[label="Clear All"]').length).toEqual(0);
  });

  it("render clear all button if there are selected categories", () => {
    const output = render({ selectedCategories: new Set(["Community"]) });

    expect(output.find('ActionButton[label="Clear all"]').length).toEqual(1);
  });
});
