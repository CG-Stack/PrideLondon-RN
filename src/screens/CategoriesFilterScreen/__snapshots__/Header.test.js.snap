// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    <ActionButton
      label="Cancel"
      onPress={[Function]}
      style={Object {}}
    />
  </View>
  <Text
    allowFontScaling={true}
    markdown={false}
    onLayout={[Function]}
    style={
      Object {
        "color": "#ffffff",
        "marginBottom": 6,
        "marginTop": 14,
      }
    }
    type="h1"
  >
    Show me
  </Text>
  <View
    style={
      Object {
        "height": 88,
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <CategoriesPills
      selectedCategories={Set {}}
      style={Object {}}
    />
  </View>
</View>
`;
